### ISSUE TITLE
Crash on postprocessing

### ISSUE THREAD

**Issue by ChWick** [OMMR4all/ommr4all-line-detection#1]
In GitLab by @chwick on Mar 15, 2019, 17:32

![crash](/uploads/49abd3ba7de43c98d8a5c8d3f4204028/crash.png)

The upper line results in the following crash:
```
Traceback (most recent call last):
  File "/home/wick/Documents/Projects/ommr4all-deploy/modules/ommr4all-server/omr/stafflines/detection/experimenter.py", line 125, in <module>
    experimenter.run()
  File "/home/wick/Documents/Projects/ommr4all-deploy/modules/ommr4all-server/omr/stafflines/detection/experimenter.py", line 116, in run
    results = list(map(run_single, train_args))
  File "/home/wick/Documents/Projects/ommr4all-deploy/modules/ommr4all-server/omr/stafflines/detection/experimenter.py", line 71, in run_single
    all_counts, all_metrics = evaluator.evaluate(args.test_pcgts_files[:1])
  File "/home/wick/Documents/Projects/ommr4all-deploy/modules/ommr4all-server/omr/stafflines/detection/evaluator.py", line 15, in evaluate
    for p in self.predictor.predict(pcgts_files):
  File "/home/wick/Documents/Projects/ommr4all-deploy/modules/ommr4all-server/omr/stafflines/detection/pixelclassifier/predictor.py", line 57, in predict
    predictions = list(self.line_detection.detect(gray_images))
  File "/home/wick/Documents/Projects/ommr4all-deploy/ommr4all-venv/lib/python3.6/site-packages/line_segmentation-0.0.1-py3.6.egg/linesegmentation/detection/lineDetection.py", line 98, in detect_basic
    yield self.detect_staff_lines(image_data)
  File "/home/wick/Documents/Projects/ommr4all-deploy/ommr4all-venv/lib/python3.6/site-packages/line_segmentation-0.0.1-py3.6.egg/linesegmentation/detection/lineDetection.py", line 139, in detect_staff_lines
    staff_list = self.postprocess_staff_systems(staff_list, staff_line_height, binarize(image_data.image))
  File "/home/wick/Documents/Projects/ommr4all-deploy/ommr4all-venv/lib/python3.6/site-packages/line_segmentation-0.0.1-py3.6.egg/linesegmentation/detection/lineDetector.py", line 302, in postprocess_staff_systems
    if image[st_point + z][i] == 0:
IndexError: index 64 is out of bounds for axis 0 with size 64
```

Parameters of the predictor:
```python
LineDetectionSettings(
    numLine=4,
    minLength=6,
    lineExtension=True,
    debug=False,
    lineSpaceHeight=20,
    targetLineSpaceHeight=10,
    model=None,
    post_process=True,
    smooth_lines=2,
    line_fit_distance=1,
)
```

Its binary and only a single line for testing purposes.

**Comment by ChWick** [comment]
In GitLab by @chwick on Mar 15, 2019, 17:33

changed the description

**Comment by ChWick** [comment]
In GitLab by @chwick on Mar 15, 2019, 18:31

changed title from **Crash** to **Crash{+ on postprocessing+}**

**Comment by ChWick** [comment]
In GitLab by @chwick on Mar 15, 2019, 18:31

changed the description

**Comment by ChWick** [comment]
In GitLab by @s331267 on Mar 15, 2019, 18:49

Crash should be fixed with ef9b3086

**Comment by ChWick** [comment]
In GitLab by @s331267 on Mar 15, 2019, 19:07

closed
